src/scripts/main.js

```diff
@@ -1 +1,75 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      resolve('First promise was resolved');
+    }
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve) => {
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      resolve('Second promise was resolved');
+    }
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    if (e.button === 2) {
+      e.preventDefault();
+      resolve('Second promise was resolved');
+    }
+  });
+});
```

Comment:
The second promise resolves on both left and right clicks. If both events occur, the promise will resolve twice, which is not ideal. Consider adding logic to ensure the promise resolves only once.